app/hotels/src/allHotels/searchForm/locationPicker/SuggestionListItem.js

```diff
@@ -43,25 +43,31 @@ export default class SuggestionListItem extends React.Component<Props> {
     this.props.onCitySelected(id, name);
   };
 
-  splitText = () => {
-    const { search } = this.props;
-    const name = idx(this.props, _ => _.city.name) || '';
-    const searchLength = search.length;
-    const matchStart = name.indexOf(search);
+  findStringToHighlight = (
+    fullText: string,
+    stringToHighlight: string,
+  ): [string, string, string] => {
+    const matchResult = fullText.match(
+      new RegExp(
+        `([\\s\\S]*?)(${stringToHighlight})([\\s\\S]*)`,
+        'i', // case insensitive
+      ),
+    );
 
-    if (!name || matchStart < 0) {
-      return { before: '', match: '', after: name };
+    if (matchResult == null) {
+      return [fullText, '', ''];
     }
 
-    return {
-      before: name.substr(0, matchStart),
-      match: name.substring(matchStart, matchStart + searchLength),
-      after: name.substring(matchStart + searchLength),
-    };
+    const [, before, match, after] = matchResult;
+    return [before, match, after];
   };
 
   render = () => {
-    const { before, match, after } = this.splitText();
+    const [before, match, after] = this.findStringToHighlight(
+      idx(this.props, _ => _.city.name) || '',
+      this.props.search,
+    );
```

Comment:
Why do you pass these strings to to class method, they are available in that method as well

Reply:
Because I believe it should be obvious what is happening - I was very confused by `this.splitString()` because the interface is not clear what string I am actually splitting. It also simplifies the test. Should I revert it?

Reply:
I like the renaming of the function and the regex. I get the benefit for the readability of the testing. 

we could do
```
findStringToHighlight = (): [string, string, string] => {
    const fullText = idx(this.props, _ => _.city.name) || '';
    const stringToHighlight = this.props.search;
...
}
```

But if you think it is best the way you have it in this PR, I am fine with it.

Reply:
@tbergq Improved.